core/common/src/rpc/web/ITwinCloudRpcProtocol.ts

```diff
@@ -0,0 +1,148 @@
+/** @packageDocumentation
+ * @module RpcInterface
+ */
+
+import { URL } from "url";
+import { BentleyStatus, Logger, OpenMode, SerializedClientRequestContext } from "@bentley/bentleyjs-core";
+import { CommonLoggerCategory } from "../../CommonLoggerCategory";
+import { IModelRpcProps } from "../../IModel";
+import { IModelError } from "../../IModelError";
+import { RpcConfiguration } from "../core/RpcConfiguration";
+import { RpcOperation } from "../core/RpcOperation";
+import { SerializedRpcOperation, SerializedRpcRequest } from "../core/RpcProtocol";
+import { RpcRequest } from "../core/RpcRequest";
+import { OpenAPIParameter } from "./OpenAPI";
+import { WebAppRpcProtocol } from "./WebAppRpcProtocol";
+
+/** An http protocol for Bentley cloud RPC interface deployments.
+ * @public
+ */
+export abstract class ITwinCloudRpcProtocol extends WebAppRpcProtocol {
+  public checkToken = true;
+
+  /** The name of various HTTP request headers based on client's request context */
+  public serializedClientRequestContextHeaderNames: SerializedClientRequestContext = {
+    /** The name of the HTTP request id header. */
+    id: "X-Correlation-Id",
+
+    /** The name of the HTTP application id header. */
+    applicationId: "X-Application-Id",
+
+    /** The name of the HTTP application version header. */
+    applicationVersion: "X-Application-Version",
+
+    /** The name of the HTTP session id header. */
+    sessionId: "X-Session-Id",
+
+    /** The name of the HTTP authorization header. */
+    authorization: "Authorization",
+
+    /** The id of the authorized user */
+    userId: "X-User-Id",
+
+    /** The content types the client can understand */
+    accept: "Accept",
+  };
+
+  /** The name of the RPC protocol version header. */
+  public protocolVersionHeaderName = "X-Protocol-Version";
+
+  /** Returns the operation specified by an OpenAPI-compatible URI path. */
+  public getOperationFromPath(path: string): SerializedRpcOperation {
+    const url = new URL(path, "https://localhost/");
+    const components = url.pathname.split("/");
+
+    const operationComponent = components.slice(-1)[0];
+    const encodedRequest = url.searchParams.get("parameters") || "";
+
+    const firstHyphen = operationComponent.indexOf("-");
+    const lastHyphen = operationComponent.lastIndexOf("-");
+    const interfaceDefinition = operationComponent.slice(0, firstHyphen);
+    const interfaceVersion = operationComponent.slice(firstHyphen + 1, lastHyphen);
+    const operationName = operationComponent.slice(lastHyphen + 1);
+
+    return { interfaceDefinition, operationName, interfaceVersion, encodedRequest };
+  }
+
+  /** Supplies the OpenAPI-compatible URI path for an RPC operation. */
+  public supplyPathForOperation(operation: RpcOperation, request: RpcRequest | undefined) {
+    const prefix = this.pathPrefix;
+    const operationName = operation.operationName;
```

Comment:
Looking at this again, I'm probably defining operation wrong here. Since this will be for making public facing RPC URLs, perhaps this should be:
`const operationId = '${operation.interfaceDefinition}-${operation.operationName}'`